app/pom.xml

```diff
@@ -193,6 +193,12 @@
             <version>4.13.2</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+            <version>3.26.3</version>
+            <scope>test</scope>
+        </dependency>
```

Comment:
Does AssertJ offer any kind of Assertions that hamcrest does not? Why do we need two assertion frameworks now?

Reply:
Hamcrest probably works fine but it wasn't used in the tests I worked with. So TBH I didn't check whether Hamcrest was already included or not and it didn't occur to me until I edited the `pom.xml`.

That being said, AssertJ's major advantange is that it does not use nested conditions and instead uses IMO a more readable syntax with actual comparing verbs like _is_, _has_, etc, while hamcrest's nesting slightly adds to the mental load of the reader. 

```java
// hamcrest
assert(actual, equalTo(ImportError.Code.UNIQUE_FIELD_ERROR));
// AssertJ
assertThat(actual).isEqualTo(ImportError.Code.UNIQUE_FIELD_ERROR);

//hamcrest
assertThat(a, nullValue());
//AssertJ
assertThat(actual).isNull();

// hamcrest
assertThat(result.getErrors(), hasSize(1));
// Assert4J
assertThat(result.getErrors()).hasSize(1);
```

I understand that hamcrest allows to wonderfully combining different assertions in one (f.i.) which in the end makes it a PITA to fix-up if the service under test changes its result structure. Most of our tests assert line by line either way.

Regarding all this I don't believe a usually professional developer can't handle two different though somewhat similar test frameworks when reading or editing tests.

Reply:
I find it highly unnecessary to use two different assertion frameworks in test. It probably will not, but it could cause confusions and I personally think it is a bit smelly. That said I do prefer AssertJ myself over Hamcrest. In then end it's probably a bigger task to remove hamcrest and work with AssertJ completely, then to just use it for the new Tests. Assuming we are moving away from Java at some point, I'm willing to just let this go.